core/actions/operation.ts

```diff
@@ -323,6 +324,16 @@ export class Operation extends ActionBuilder<dataform.Operation> {
     return dataform.Target.create(this.proto.target);
   }
 
+  /** @hidden */
+  public getHasOutput() {
+    return this.proto.hasOutput;
+  }
+
+  /** @hidden */
+  public setFilename() {
+    return this.proto.hasOutput;
```

Comment:
This is very counterintuitive. Setter in fact sets nothing. What was the intention here?

Reply:
This method is totally wrong, and it works because it's not being used. I've realized we don't actually need to expose a `setFilename` for most classes, so have removed and updated actions to pass is in via configs